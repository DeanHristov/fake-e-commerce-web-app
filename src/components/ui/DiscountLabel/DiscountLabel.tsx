import { FC } from 'react';

export interface IDiscountLabelProps {
  discount: number;
}

const DiscountLabel: FC<IDiscountLabelProps> = ({ discount }) => {
  return (
    <div className="card-flag-container">
      <span className="card-flag">{`-${discount}%`}</span>
    </div>
  );
};

export default DiscountLabel;
