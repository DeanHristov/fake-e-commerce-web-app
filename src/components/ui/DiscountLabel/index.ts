import DiscountLabel, { IDiscountLabelProps } from './DiscountLabel';

export type { IDiscountLabelProps };
export default DiscountLabel;
