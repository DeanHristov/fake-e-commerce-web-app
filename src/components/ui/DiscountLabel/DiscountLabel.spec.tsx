import { render } from '@testing-library/react';
import DiscountLabel from './';

describe('UI/Component <DiscountLabel {...} />', () => {
  it('Should be able to render the default state', () => {
    const { container } = render(<DiscountLabel discount={20} />);

    expect(container).toMatchSnapshot();
  });
});
