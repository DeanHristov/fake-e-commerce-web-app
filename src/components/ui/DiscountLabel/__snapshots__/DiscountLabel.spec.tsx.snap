// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Component <DiscountLabel {...} /> Should be able to render the default state 1`] = `
<div>
  <div
    class="card-flag-container"
  >
    <span
      class="card-flag"
    >
      -20%
    </span>
  </div>
</div>
`;
